src/include/sof/audio/audio_stream.h

```diff
@@ -515,8 +537,18 @@ audio_stream_get_free_samples(const struct audio_stream __sparse_cache *stream)
 static inline uint32_t
 audio_stream_get_free_frames(const struct audio_stream __sparse_cache *stream)
 {
-	return audio_stream_get_free_bytes(stream) /
-		audio_stream_frame_bytes(stream);
+	uint32_t frame_bytes = audio_stream_frame_bytes(stream);
+
+	switch (frame_bytes) {
+	case 2:
+		return audio_stream_get_free_bytes(stream) >> 1;
+	case 4:
+		return audio_stream_get_free_bytes(stream) >> 2;
```

Comment:
we'd better try to save switch, simply:
audio_stream_get_free_bytes(stream) >> (frame_bytes >> 1)